internal/app/machined/pkg/controllers/network/hostname_config.go

```diff
@@ -172,6 +209,21 @@ func (ctrl *HostnameConfigController) apply(ctx context.Context, r controller.Ru
 	return ids, nil
 }
 
+func (ctrl *HostnameConfigController) getStableDefault(machineID []byte) (*network.HostnameSpecSpec, error) {
+	id := string(machineID)
+
+	if len(id) < 6 {
+		return nil, fmt.Errorf("machine ID is too short: %s", id)
+	}
+
+	hostname := "talos-" + id[0:6]
```

Comment:
should we make it easier for users to figure out which hostname has which ip by storing this data in a resource maybe, nothing as a blocker for this PR, just some ideas

Reply:
Hm, I'm not sure if it is worth it tbh - if the cluster is there, they can do `kubectl get node -owide`. If not, bunch of `talosctl get`s would be enough. @smira any inputs?

Reply:
I don't think we can make it any easier... you can do `taloctl get members` to see the mapping